src/Lean/Data/HashMap.lean

```diff
@@ -140,11 +150,16 @@ def erase [BEq α] [Hashable α] (m : HashMapImp α β) (a : α) : HashMapImp α
     if bkt.contains a then ⟨size - 1, buckets.update i (bkt.erase a) h⟩
     else m
 
-inductive WellFormed [BEq α] [Hashable α] : HashMapImp α β → Prop where
-  | mkWff          : ∀ n,                    WellFormed (mkHashMapImp n)
+@[inline] def map (f : β → γ) (m : HashMapImp α β) : HashMapImp α γ :=
+  { size := m.size, buckets := m.buckets.mapVal f }
+end
+
+inductive WellFormed {α : Type u} [BEq α] [Hashable α] : {β : Type v} → HashMapImp α β → Prop where
+  | mkWff          : ∀{β : Type v} n, WellFormed (@mkHashMapImp α β n)
   | insertWff      : ∀ m a b, WellFormed m → WellFormed (insert m a b |>.1)
   | insertIfNewWff : ∀ m a b, WellFormed m → WellFormed (insertIfNew m a b |>.1)
   | eraseWff       : ∀ m a,   WellFormed m → WellFormed (erase m a)
+  | mapWff         : ∀(f : β → γ) m, WellFormed m → WellFormed (map f m)
```

Comment:
This doesn't work, it has universe issues. I've lost track of whether this code is now also used in Std.HashMap, but we can't do this in that case. For `Lean` it doesn't really matter since we can probably set all the universes to 0 anyway.